pkg/k8s/resource/resource.go

```diff
@@ -822,3 +825,33 @@ func getUID(obj k8sRuntime.Object) types.UID {
 	}
 	return meta.GetUID()
 }
+
+type WorkqueueMetricsProvider struct{}
+
+func (WorkqueueMetricsProvider) NewDepthMetric(name string) workqueue.GaugeMetric {
+	return metrics.WorkQueueDepth.WithLabelValues(name)
```

Comment:
Won't this global just always be Noop?  Trying this out locally this seems to break the metrics as these are not part of the registry being used to gather metrics.